src/exposition/http.rs

```diff
@@ -107,6 +122,58 @@ mod handlers {
                         }
                     }
                 }
+                if config.prometheus().histograms() {
+                    if let Some(snapshot) = snapshots.previous.get(metric.name()) {
+                        // downsample the snapshot if necessary
+                        let downsampled: Option<Snapshot> =
+                            if config.prometheus().histogram_downsampling_factor() > 0 {
+                                Some(
+                                    snapshot
+                                        .downsample(
+                                            config.prometheus().histogram_downsampling_factor(),
+                                        )
+                                        .unwrap(),
+                                )
+                            } else {
+                                None
+                            };
+
+                        // reassign to either use the downsampled snapshot or the original
+                        let snapshot = if let Some(snapshot) = downsampled.as_ref() {
+                            snapshot
+                        } else {
+                            snapshot
+                        };
+
+                        // we need to export a total count (free-running)
+                        let mut count = 0;
+                        // we also need to export a total sum of all observations
+                        // which is also free-running
+                        let mut sum = 0;
+
+                        let mut entry = format!("# TYPE {name}_distribution histogram\n");
+                        for bucket in snapshot {
+                            // add this bucket's sum of observations
```

Comment:
Is it common to export bucket values as a count instead of a gauge (after delta)? Percentiles are always exported as gauges, while raw buckets can be exported as either, gauges seem more convenient at query time.

Reply:
It appears that is how the bucket values are expected to behave: https://prometheus.io/docs/concepts/metric_types/#histogram

It does allow easier downsampling in the time domain if the TSDB also uses free-running counters.

Reply:
Gotcha. This wasn't my intuition but I see the point.